core/domain/collection_domain.py

```diff
@@ -992,6 +1167,38 @@ def validate(self, strict=True):
         for node in self.nodes:
             node.validate()
 
+        # Validate skills.
```

Comment:
What about validating ``next_skill_id`` itself? Also, are there tests for each of these new validation errors?